### ISSUE TITLE
[Bug] XAML Hot Reload not working on Windows

### ISSUE THREAD

**Issue by dickulus** [dotnet/maui#3369]
### Description

XAML Hot Reload and Live Preview not working on Windows, even though they work fine on Android. Tested on latest VS 2022 17.1 preview 1 and MAUI workload installed by the Visual Studio Installer

I tried uninstalling and reinstalling everything .NET and VS related like a billion times, didn't help.

### Steps to Reproduce

1.  Create new MAUI sample app
2. Run
3. Try changing something, like the text in my screenshots

### Expected Behavior

XAML Hot reload working, like in every MAUI promotional video made in past several months...

### Actual Behavior

XAML Hot reload not working

### Basic Information

- Version with issue: latest installed by Visual Studio installer
- IDE: Visual Studio Enterprise 2022 17.1. Preview 1
- Platform Target Frameworks:
  - UWP:  net6.0-windows10.0.19041

### Screenshots

Hot reload enabled in options
<img width="537" alt="Screenshot 2021-11-12 19 42 18" src="https://user-images.githubusercontent.com/16528222/141529701-8214972f-735d-4825-9c3b-6bef9521445e.png">

Nothing happens after XAML file changes and gets saved, even though Hot Reload should update the app
<img width="942" alt="Screenshot 2021-11-12 19 43 07" src="https://user-images.githubusercontent.com/16528222/141529710-61d0dd2b-d8d1-4cef-ac65-ff96bf63f3ff.png">

"XAML Live Preview unavailable" when it definitelly should be available
<img width="941" alt="Screenshot 2021-11-12 19 45 00" src="https://user-images.githubusercontent.com/16528222/141529717-a16a2f88-ef6d-40fb-b51f-ad3afa56d544.png">

**Comment by csulmone** [comment]
I am getting this same issue as well on VS 2022 17.1 preview 1

**Comment by Wowotoe** [comment]
I'm getting the same result too. No hot reload with default MAUI app when targeting Windows. But she did the demo with VS 17.1 preview 1, supposedly.

https://youtu.be/LUlbO5MjPjU

**Comment by ArieBen** [comment]
Anyone found a workaround?
I was eager to start a  MAUI Blazor app. ==> "XAML Hot reload unavailable".
razor pages changes can be reloaded with a save + an f5 on the app but it won't reflect any css changes, which makes the whole thing unusable.

**Comment by CliffAgius** [comment]
This issue is still present I just installed a fresh VS22 17.1 Preview 1.1 and with a File->New MAUI app Hot Reload doesn't work get the following error in my sample App:

    

>  Caught exception in AgentStatusChangedHandler at 313: System.Exception: One or more errors occurred. (Could not load type of field 'Xamarin.HotReload.Xaml.VisualTreeService.VisualTree.ApplicationWrapper:dispatcher' (0) due to: Could not resolve type with token 01000045 from typeref (expected class 'Microsoft.Maui.Controls.IDispatcher' in assembly 'Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null') assembly:Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null type:Microsoft.Maui.Controls.IDispatcher member:(null))
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)

I will submit a `Report a Problem` ticket as well as I don't think it's only MAUI this issue is affecting.

**Comment by dickulus** [comment]
Still broken in VS 2022 17.1 preview 2

**Comment by nZeus** [comment]
> Still broken in VS 2022 17.1 preview 2 by default but I found a fix: the tutorial tells you to just install the Mobile Development workload in VS Installer, but you also have to install UWP and Desktop C++ workloads, or at least that was what fixed it for me.

I just tried this - didn't help :(

**Comment by drakepro** [comment]
Same problem here with VS 2022 17.1 preview 2.  Workaround - Navigating or button clicks seem to re-render the UI with your change.  In the default application template, go to the "Counter" page, make a change in the HTML, save your change, then click the "Click me" button and then your change will be visible.

I will try @dickulus's suggestion and see if it works for me.

**Comment by ychost** [comment]
> Still broken in VS 2022 17.1 preview 2 by default but I found a fix: the tutorial tells you to just install the Mobile Development workload in VS Installer, but you also have to install UWP and Desktop C++ workloads, or at least that was what fixed it for me.

thanks, it works for me

**Comment by RyanThiele** [comment]
With the android emulator - after installing the workloads suggested:

- .NET desktop development
- Desktop development with C++

I was able to use Hot-Reload. When I loaded a new MAUI project without closing the emulator I lost hot reload, even after restarting the project. 
I was able to re-enable Hot-Reload after restarting the emulator. I also re-ran the application and Hot-Reload was reconnected and still working.
